src/protocolgame.cpp

```diff
@@ -988,6 +988,26 @@ void ProtocolGame::parseRotateItem(NetworkMessage& msg)
 	addGameTaskTimed(DISPATCHER_TASK_EXPIRATION, &Game::playerRotateItem, player->getID(), pos, stackpos, spriteId);
 }
 
+void ProtocolGame::parseRuleViolationReport(NetworkMessage &msg)
+{
+	RuleViolationType_t report_type = (RuleViolationType_t) msg.getByte();
+	RuleViolationReasons_t report_reason = (RuleViolationReasons_t) msg.getByte();
```

Comment:
To add on to this, since you can infer the type from the cast, you can use `auto` here.
`auto reportType = static_cast<RuleViolationType_t>(msg.getByte());`

However, you shouldn't cast the byte for a couple of reasons.
1. If the enum doesn't contain the value then that could lead to undefined behavior.
2. `if (reportType == REPORT_TYPE_NAME)` will work whether you cast or not. Even though @ranisalt mentioned not forcing enum type, it may be good to change it to `uint8_t` to match the value returned from `getByte()`.